spec.bs

```diff
@@ -980,6 +980,77 @@ To <dfn>add a part</dfn> given a [=pattern parser=] |parser|, a string |prefix|,
+      <p class=note>We collect the list of matching group names in a parallel list.  This is largely done for legacy reasons to match [path-to-regexp](https://github.com/pillarjs/path-to-regexp).  We could attempt to convert this to use regular expression named captured groups, but given the complexity of this algorithm there is a real risk of introducing unintended bugs.  In addition, if we ever end up exposing the generated regular expressions to the web we would like to maintain compability with [path-to-regexp](https://github.com/pillarjs/path-to-regexp) which has indicated its unlikely to switch to using named capture groups.
+    1. Let |regexp value| be |part|'s [=part/value=].
+    1. If |part|'s [=part/type=] is "<a for=part/type>`segment-wildcard`</a>", then set |regexp value| to the result of running [=generate a segment wildcard regexp=] give |options|.
+    1. Else if |part|'s [=part/type=] is "<a for=part/type>`full-wildcard`</a>", then set |regexp value| to [=full wildcard regexp value=].
+    1. If |part|'s [=part/prefix=] is the empty string and |part|'s [=part/suffix=] is the empty string:
+      <div class=note>
+      <p>If there is no [=part/prefix=] or [=part/suffix=] then we generate a simple capturing group.  It uses the form:
+      <p>`(<regexp value>)<modifier>)`.
+      </div>
+      1. Append "`(`" to the end of |result|.
+      1. Append |regexp value| to the end of |result|.
+      1. Append "`)`" to the end of |result|.
+      1. Append the result of running [=convert a modifier to a string=] give |part|'s [=part/modifier=] to the end of |result|.
+      1. [=Continue=].
+    1. If |part|'s [=part/modifier=] is "<a for=part/modifier>`none`</a>" or "<a for=part/modifier>`optional`</a>":
+      <div class=note>
+      <p>This section handles non-repeating parts with a [=part/prefix=] and/or [=part/suffix=].  There is an inner capturing group that contains the primary |regexp value|.  The inner group is then combined with the [=part/prefix=] and/or [=part/suffix=] in an outer non-capturing group.  Finally the modifier is applied.  The overall form is:
+      <p>`(?:<prefix>(<regexp value>)<suffix>)<modifier>)`.
+      </div>
+      1. Append "`(?:`" to the end of |result|.
+      1. Append the result of running [=escape a regexp string=] given |part|'s [=part/prefix=] to the end of |result|.
+      1. Append "`(`" to the end of |result|.
+      1. Append |regexp value| to the end of |result|.
+      1. Append "`)`" to the end of |result|.
+      1. Append the result of running [=escape a regexp string=] given |part|'s [=part/suffix=] to the end of |result|.
+      1. Append "`)`" to the end of |result|.
+      1. Append the result of running [=convert a modifier to a string=] give |part|'s [=part/modifier=] to the end of |result|.
+      1. [=Continue=].
+    1. [=Assert=]: |part|'s [=part/modifier=] is "<a for=part/modifier>`zero-or-more`</a>" or "<a for=part/modifier>`one-or-more`</a>".
```

Comment:
Heh I see what you're doing here with the asserts.

I think it'd be a little more convention to have an "Else" block on this line and add the asserts indented under the else.

Reply:
And drop the continue in the previous section?

Reply:
I accidentally indented things here.  I'll un-indent.